Show chat name, source contact and time in PN body

pedropombeiro opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)
This PR adds some needed context to desktop PNs:

![image](https://user-images.githubusercontent.com/138074/48507577-9af97b80-e84c-11e8-88ff-fea76f385b08.png)

It is a simple change with no implications on other components.

### Testing notes:
<!-- (Specify which platforms should be tested) -->
#### Platforms
- macOS
- Linux
- Windows

<!-- (Specify if some specific areas has to be tested, for example 1-1 chats) -->
#### Areas that maybe impacted (optional)
**Functional**
- 1-1 chats
- public chats
- group chats

status: ready <!-- Can be ready or wip -->

Commit:
Show chat name, source contact and time in PN body. Part of #6621

- Add higher priority for 1:1 messages

Signed-off-by: Goran Jovic <goranjovic@gmail.com>

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

EugeOrtiz commented:
@PombeirP Can you provide examples of how they will look for 1:1 chat messages and replies in a public channel? 
I wouldn't worry about general public channel PNs, since [we want to limit them](https://github.com/status-im/status-react/issues/6445).
In any case, you can see the proposed [PNs format here.](https://github.com/status-im/status-react/issues/6621)

pedropombeiro commented:
There are a few considerations done in the logic:
- Timestamp is only shown if message is older than 15 seconds
- Contact name is only shown if different than chat name (i.e. multi-user chats)

pedropombeiro commented on src/status_im/chat/models/message.cljs:
```diff
@@ -100,7 +100,11 @@
                (get-in db [:account/account :desktop-notifications?])
                (< (time/seconds-ago (time/to-date timestamp)) constants/one-earth-day))
-      (.sendNotification react/desktop-notification (:text content)))
+      (let [chat-name     (get-in db [:chats chat-id :name])
```
@yenda I guess I should use `:get-chat-name` here?

churik commented:
Amazing work @PombeirP!

Some issues:
1) Cannot see three-random-name of person who is not added to your contact list (Mac Os, Linux)
<img width="165" alt="pns 1" src="https://user-images.githubusercontent.com/4557972/48559080-275a7b80-e8eb-11e8-9e8a-f9a61a328f67.png">

2) On Linux - time is missing on PNs
<img width="187" alt="linux_time" src="https://user-images.githubusercontent.com/4557972/48562107-cedaac80-e8f1-11e8-9f3a-6fad7b592f58.png">

3) As for me would be nice to have the same format in both platform.
For now:
<img width="193" alt="linux_2" src="https://user-images.githubusercontent.com/4557972/48563079-d26f3300-e8f3-11e8-95e7-7f41bff973ca.png">
I would prefer to have the same view for Linux as it is for Mac OSx.

and separate thank you for nice icon on linux!

pedropombeiro commented:
Still waiting for input from @yenda about that. Time is not missing, see the 15 sec rule.

churik commented:
@PombeirP I saw it, but it works differently on Linux and Mac OSx.
On last screen, you can see that the same message for same account contains a timestamp for Mac OSx and doesn't for Linux (I had opened app for Mac OSx and for linux VM at the same time to compare it)
For Mac OSx as Ia can observe timestamps are shown regardless rule of 15 sec.

pedropombeiro commented:
> and separate thank you for nice icon on linux!

My pleasure, I'm a Linux user and it had been bothering me for a long time!

> I would prefer to have the same view for Linux as it is for Mac OSx. 

I don't understand, there is absolutely nothing in the code that could differentiate between platforms, all 3 desktop platforms run exactly the same code. Since I run Linux, I know that it is showing for me. Unless there is some platform-specific bug in `(time/seconds-ago (time/to-date timestamp))`. Any clues @yenda?

@churik please do one test, sign off the account, send multiple messages from another device to it, wait 15 secs, then sign in and minimize the app. All the notifications should have the time on them.

pedropombeiro commented:
@churik From what I can understand from the code, it looks like we'll need to do a small refactoring to make it possible to show the anonymous name in the notification, so we have two options: we leave this open until someone from the Clojure team can do that refactoring, or we merge and file a bug against this behavior. Either way is fine by me.

pedropombeiro commented:
@yenda tells me that his branch already contains the change required for this, so maybe he could fix this there after this is merged to `develop`.

pedropombeiro commented:
@churik a new build is underway which fixes the missing contact name.

pedropombeiro commented:
- Refactored the building of the desktop notification out of the `add-message` method for more clarity/separation of concerns. Would appreciate a second review.
- Added support for notification priority. 1:1 messages will have higher priority than others.

pedropombeiro commented:
OK, @churik build #10 should be OK to test. I had to remove markup since some platforms showed the HTML tags

churik commented:
@pablodip 
now it is working exactly as you described.
Thank you again.
Ready to merge.
